src/lib/backendHelper.js

```diff
@@ -0,0 +1,44 @@
+import { get } from "lodash";
+import { fromJS } from "immutable";
+import { fileExtension } from "Lib/pathHelper";
+import unsentRequest from "Lib/unsentRequest";
+import { CURSOR_COMPATIBILITY_SYMBOL } from "../valueObjects/Cursor";
+
+export const filterByPropExtension = (extension, propName) => arr =>
+  arr.filter(el => fileExtension(get(el, propName)) === extension);
+
+const catchFormatErrors = (format, formatter) => res => {
+  try {
+    return formatter(res);
+  } catch (err) {
+    throw new Error(`Response cannot be parsed into the expected format (${ format }): ${ err.message }`);
+  }
+};
+
+const responseFormatters = fromJS({
+  json: async res => {
+    const contentType = res.headers.get("Content-Type");
+    if (!contentType.startsWith("application/json") && contentType.startsWith("text/json")) {
```

Comment:
This can be switched to just check for `startsWith('text/json')`.

Reply:
It should actually be `!contentType.startsWith("application/json") && !contentType.startsWith("text/json")`, but the second `!` was missing. Parsing a `Content-Type` of `"text/json"` as JSON shouldn't produce an error.